### DISCUSSION TITLE
What is Nostr, and how to start using Nostr

### DISCUSSION THREAD

**Discussion by vishalxl** [vishalxl/nostr_console#31] [General]
Nostr is a protocol built using self-owned account/identities so that it enables us to create 'decentralized' social networks and even other types of solutions. Its [project page](https://github.com/nostr-protocol/nostr) has more technical details. And [the 'awesome nostr' page](https://github.com/aljazceru/awesome-nostr) lists almost all current clients and applications, but its also quite a large list. This particular page will give a brief overview of Nostr and how to use it. 

# Nostr Social Network

The Nostr social network is build much like twitter, where you can create posts ( like tweets), like posts, follow someone or unfollow them, retweet/repost. Normally the term 'post' or 'note' is used to refer to creating a post on the nostr social network. 

# How to Start using Nostr

Using Nostr means using one of its clients to access the Nostr network. We will go over the clients, and then how you can use the clients. 

## Nostr clients 

Some of more popular nostr clients are:

1. [astral.ninja](https://astral.ninja) , a nostr web client which has twitter-like social network ( which is also typically called the 'nostr' network)
2. [coracle.social](https://coracle.social) has support for public channels.
3. Damus is a MacOS and iPhone app that's available on Apple Testflight. It supports twitter like social network, as well as shows messages from channels too. 
4. Nostr Console is another client which is more for people familiar with command line. It's a terminal-UI based client, and has binaries for windows, linux and macOS. 
5. A larger comparison list is [here](https://github.com/vishalxl/Nostr-Clients-Features-List). 
6. Almost all clients support encrypted direct messaging to other users. 

## Steps to use Nostr

1. **Create a private/public keypair** using a nostr client like [astral.ninja](https://astral.ninja). The client will ask you whether you want to create such a key pair, and if you agree, it will create that for you. This is used as a 'self-sovereign' identity in the Nostr system. 
2. **Save the private key**: Once you have created a private/public key pair, you should keep or save the private key somewhere, and keep it secret. Also, if you lose your newly created private key, you can always generate a new one, however you may lose access to your old private messages and such, but not public messages. 
4. **Create multiple keys if needed**: The same key can be used on all nostr clients. If you want to create multiple identities or accounts, you can create new such key pairs. 
5. **Update your profile**: From the client you're using, you should be able to update your profile name, picture etc. This depends on the client whether and how much you can update. Typically you can add your name, about me section and profile picture.
8. **Make posts**: The client should also show you some posts by others. Also it should allow you to create posts, follow others. ( astral damus etc clients do allow you to follow others; anigma.io doesn't since its like telegram and has a public chat rooms). 
9. **Find users**: Astral.ninja has a global feed where you can see posts by others. The page [nostr.io/stats](https://nostr.io/stats) shows some of the more prolific users. The page [damus.io/channels](https://damus.io/channels) shows some of the more active public channels. 

**Note** about key formats: Damus, and astral use npub/nsec key format, while other clients use hex. To convert from npub/nsec to hex format, use [this tool](https://damus.io/key). Hex keys will be phased out of all user-facing parts of the clients. 

# Contact and More Info

* Nostr [telegram group](https://t.me/nostr_protocol)
* Nostr public channel id: 25e5c82273a271cb1a840d0060391a0bf4965cafeb029d5ab55350b418953fbb
* on [well ordrered wiki](https://wiki.wellorder.net/post/nostr-intro/), an introduction to Nostr
* slightly more technical article on what is Nostr and [how Nostr works](https://gist.github.com/r3nor/93cfbfe76c983a701321d3e30c990174)

**Comment by sheldonkotyk** [comment]
The link to damus.io/channels is broken. There's a . after io instead of /

**Reply by vishalxl** [reply]
Fixed, thanks.

**Comment by treyda6** [comment]
When editing profile, what does it mean to "paste the signature here (as hex)"?

**Reply by vishalxl** [reply]
hey, which client are you using?

**Reply by treyda6** [reply]
astral

**Reply by vishalxl** [reply]
Looks like you logged in using your public key, rather than private key.

Can you make any posts? If you can't then that means you used your public key.

?

**Reply by rodrigocipriani** [reply]
Going through the same thing with astral, I'm starting just with the public key, and when I try to post something, it asks for a signature, because I've not storage my private key, but I've no clue how to create that signature, I've even tried with Schnorr.
Same thing on nostr.rocks

The image shows what I'm saying. 
![image](https://user-images.githubusercontent.com/1211294/208795335-322c1807-f7a1-47cb-88df-91a23a6e607e.png)

**Reply by vishalxl** [reply]
Haven't seen this in a while, but you can ignore this. It seems its asking the user to sign the event using some command line tool like nostril, and then paste the signature from there. You should be able to use getalby extension here rather than doing this. Which will sign your events for you.

**Reply by rodrigocipriani** [reply]
Thanks for answering. The thing is you'll see it if you use just the public key. I was trying to figure out how to generate this key without using any extension.

**Comment by siniradam** [comment]
Can someone explain how Private & Public Key pair works?
I use same private key, but seems like, each platform generates another public key.

I'm not sure if I get the logic behind this. astral.ninja providing only Private or Public KEY Option, yet I can pick both on anigma.io

What is happening when I have different public keys with the same private key?

**Reply by vishalxl** [reply]
One specific pvt key gets you the same public key everytime; in other words a private key does not give you multiple different public keys. 

Astral allows you to 'log' in using both a pvt key and public key. Typical use is to log in using pvt key, because then you can create posts. but when you 'log' in using just the public key, then you can view the site as that user, but can't send any posts.

On Anigma.io, you can only log in using private key, and not using public key.

**Reply by siniradam** [reply]
So what is happened, I tried to login astral.ninja first, Clicked to generate, while public was selected. Couldn't change later on, went to settings to find out, there wasn't any private either So It's weird to generate a public key without private. 
So went to Anigma which generated both keys simultaneously, then I went back to Astral signed out, logged in with the same private key, but public key was different on the settings page. This is what confused me.

**Reply by cogentgene** [reply]
>but when you 'log' in using just the public key, then you can view the site as that user, but can't send any posts.
What is the point of this? Seems like poor UX?

**Reply by vishalxl** [reply]
> > but when you 'log' in using just the public key, then you can view the site as that user, but can't send any posts.
> > What is the point of this? Seems like poor UX?

Its useful for some, because there are browser extensions that act as your private key store, and when you do not log in to a client like astral.ninja, the client may ask the extension to sign events on your behalf. then the client does not need your private key. 

Which is excellent safety, because you are not giving your private key to random online clients.

**Reply by ckapilla** [reply]
astral.ninja seems to be completely broken in that it doesn't create a private key at all. That can't be right.

**Reply by ckapilla** [reply]
so then I tried Anigma (thanks for the tip) and got both public and private keys. But then it completely locked up my browser (Brave) and I had to kill the process using task manager. But after restarting astral.ninja was showing both the original public and private keys. Confusing, you say???

**Reply by siniradam** [reply]
> so then I tried Anigma (thanks for the tip) and got both public and private keys. But then it completely locked up my browser (Brave) and I had to kill the process using task manager. But after restarting astral.ninja was showing both the original public and private keys. Confusing, you say???

Unfortunately astral.ninja is hogging immense amount of system resources. I thought my macbook is going to take of and leave the house. Since both chrome and brave are using same engine (chromium) result is the same, on safari it is not working at all. Haven't tried on anything else yet.

I'm still trying to figure out the (nostr) system, once I do I'll release my own client with some local db features, thinking PouchDB.

So far I've designed something like this. But I need to figure out to make it work with svelte.
https://www.figma.com/file/QdL2SoQ0i3TvZ5b7Rw437x/NOSTR-%7C-talking.fyi?node-id=1%3A2&t=v2xEZAV4ul2rHMLV-1

**Comment by gap892003** [comment]
Can I please suggest removing Twitter references here? 
For nontechnical users its fine, but here we can ommit that.

**Reply by Mrydnn** [reply]
OMIT

**Comment by drscotthawley** [comment]
One of the profile prompts is for a "NIP-05 Identifier": What is that? 

(was randomly assigned to nostr.rocks by nostr.directory).

**Reply by vishalxl** [reply]
hey, you can ignore NIP 05 if you want; not everybody uses it nor need to. Its for people who own a domain/website and want to 'prove' they own it on nostr so their identity can be 'verified'.

**Reply by Mrydnn** [reply]
Thank you! I also had that question. I assumed that was what it was, but it's nice to have confirmation.

**Comment by metasikander** [comment]
maybe you want to add my guide for setting up nip-5?
it might be too out of scope of this getting started guide, but maybe something for people that want to go further?
https://gist.github.com/metasikander/609a538e6a03b2f67e5c8de625baed3e or https://lhote.me/posts/nip05-guide/

**Reply by vishalxl** [reply]
If anyone has time, they can convert [this ](https://twitter.com/0xsommerfeld/status/1604795583755161602)into a discussion or page anywhere.

**Reply by vishalxl** [reply]
hey metasikander, can you also add steps to use github there? that's also a good way to get verification for majority of people.

**Reply by metasikander** [reply]
mmm. good point. i have a busy pre-christmas period right now, but i'll add it as soon as i can!

**Reply by metasikander** [reply]
Done! :smile: 
https://lhote.me/posts/nip-05_github_guide/

**Reply by vishalxl** [reply]
> Done! 😄 https://lhote.me/posts/nip-05_github_guide/

Great guide! one thing: Should mention [.nojekyll](https://github.com/vishalxl/vishalxl.github.io/blob/main/.nojekyll) because as i understand thats needed to serve files in this scenario.

**Reply by metasikander** [reply]
Thank you!

Are you sure?
i can add it but it looks like it works without it with the repo i used: https://metasikander.github.io/nip05-github/.well-known/nostr.json

**Reply by vishalxl** [reply]
> Thank you!
> 
> Are you sure? i can add it but it looks like it works without it with the repo i used: https://metasikander.github.io/nip05-github/.well-known/nostr.json

Ok, i see. I'm not sure.. but was under that impression.

**Reply by metasikander** [reply]
probably doesn't hurt, so i'll add it anyway :relaxed:

**Comment by aburra16** [comment]
Is it possible(or meaningful) to create a "relay-of-relays"? In other words, create a relay hub or aggregator - that way, users can add just one or two big ones, and that covers their events being published across multiple relays (and therefore clients)?

**Reply by vishalxl** [reply]
Higher level solutions, and improvements in protocols, will almost inevitably happen. So people could go along this way too, but right now no one can be sure what exact forms they'll take...

**Comment by offordscott** [comment]
The Damus TestFlight is full. 🥺

**Reply by stephenmontague** [reply]
Thanks for confirming this. That’s sad

**Comment by siniradam** [comment]
ELI5 guide 
https://gist.github.com/siniradam/73cf670871228daeaeeb7593c6d26999

**Comment by lucash-dev** [comment]
How would I go about setting up a relay that can scale to 1M daily active users or more?
Is there any implementation right now likely to work? 
If not, any suggestions on an implementation that could be a reasonable starting point?

**Reply by vishalxl** [reply]
hey, best place to ask this is the telegram group mentioned above.

**Comment by ptrio42** [comment]
Would you consider adding this basic newcomers guide to 'More Info' section? https://uselessshit.co/resources/nostr/

**Comment by ucanozen** [comment]
Minor grammar error: "It's" in "It's [project page](https://github.com/nostr-protocol/nostr) has" should be "Its".

**Comment by Cryptolyfe** [comment]
Does anyone have any idea how many users this app has?  Rough estimate?

**Reply by oqnoj** [reply]
![image](https://user-images.githubusercontent.com/74079707/215115580-2fe890c8-f777-40f2-a807-dde36a864cef.png)
source is npub19mduaf5569jx9xz555jcx3v06mvktvtpu0zgk47n4lcpjsz43zzqhj6vzk

**Reply by oqnoj** [reply]
or here https://nostr.io/stats

**Comment by Highrigh6677** [comment]
Can anyone explain why everyone is blocked for me?

**Reply by vishalxl** [reply]
What client are you using? ( Nostr is just a protocol, but your 'client' will define what your experience is. )

**Reply by StellarStoic** [reply]
Check which relays your client uses. If there are none, add a few ... https://nostr.watch/relays/find

**Comment by greg121** [comment]
Damus now in Appstore

**Comment by CoNETProject** [comment]
Great idea.
It has a huge problem is privacy.

E2E networking based on IP address will directly expose the connection between the user's key and IP address, thereby exposing the user's identity.

This is why SEGURO uses CoNET Decentralized Stealth Internet Infrastructure (IP-less network)

https://user-images.githubusercontent.com/19976150/216024685-ec2cf942-89ca-496b-82fe-e92e67abf118.mp4

**Comment by earonesty** [comment]
how do you link to a note?  is it like # noteid or @ noteid or something?

**Reply by verityj** [reply]
Figured it out. Inside your note you write nostr: followed by the note code. Example: `nostr:note1z7lysage232ukggx69plhc2gr7y63zvech3grm9td6033w3tetzsrh640q` this would link to that specific note of mine. Similar to tagging `@username` with `nostr:npub....` inside a post (if the user is not part of your network, it may not pop up if you type `@username` - depends on nostr app or site).
